connector/google/google.go

```diff
@@ -40,12 +40,16 @@ type Config struct {
 	// If this field is nonempty, only users from a listed group will be allowed to log in
 	Groups []string `json:"groups"`
 
-	// Optional path to service account json
+	// Optional path to service account JSON
 	// If nonempty, and groups claim is made, will use authentication from file to
 	// check groups with the admin directory api
 	ServiceAccountFilePath string `json:"serviceAccountFilePath"`
 
-	// Required if ServiceAccountFilePath
+	// Recommended when running inside Google Cloud Platform
+	// Use this instead of ServiceAccountFilePath to use default credentials
+	UseGoogleDefaultCredentials bool `json:"useGoogleDefaultCredentials"`
```

Comment:
I wonder if impersonation should be the default instead when service account file is not provided.

Reply:
cc @JoelSpeed

Reply:
If I understand your comment correctly, we cannot directly interact with GSuite Admin API with service accounts without a GSuite account impersonation. So regardless of the authentication method (key or no key), we will require to go through impersonation step.

Reply:
@NaurisSadovskis - did you test this out? We are looking to use workload-identity instead of long-lived keys. when can we expect this to rollout?

Reply:
@NaurisSadovskis - any progress on this?